### ISSUE TITLE
Model not getting trained on single GPU

### ISSUE THREAD

**Issue by aryanmangal769** [yrcong/RelTR#32]
When I try to train on single GPU, the error keeps on increasing and I cannot see any good results even thill 38th epoch. 

train_class_error starts from 97.88 and deom 19th to 37th epoch its consistently 100.  Can you debug this?

Please let me know if you need some more information

**Comment by yrcong** [comment]
We train the model for 150 epochs. 38th epoch might be just warm-up. Maybe you can try to load some pretrained weights to accelerate the training?

**Comment by qql-Kenneth** [comment]
@aryanmangal769 bro! How to train the model on One GPU?

**Comment by qql-Kenneth** [comment]
@me I add `os.environ['MASTER_PORT'] = '8889' ` in main.py

**Comment by yrcong** [comment]
It is not related to the port. Make --nproc_per_node=1 pls

**Comment by AlphaGoooo** [comment]
I trained 70 epochs but the results are still bad, including the errors and the loss. The loss is always like 33, 34..., is it normal or something goes wrong?
![微信图片_20240712155952](https://github.com/user-attachments/assets/fae57e08-5c87-4aab-8f73-e54bea819b4c)

**Comment by wuzhiwei2001** [comment]
> It is not related to the port. Make --nproc_per_node=1 pls

I set --nproc_per_node=1, but I am still getting the error torch.distributed.elastic.multiprocessing.errors.ChildFailedError. How can I resolve this issue?thank your reply

**Comment by A11en4z** [comment]
> > It is not related to the port. Make --nproc_per_node=1 pls
> 
> I set --nproc_per_node=1, but I am still getting the error torch.distributed.elastic.multiprocessing.errors.ChildFailedError. How can I resolve this issue?thank your reply

Maybe you should update the torch version, torch 1.6+cuda10.1 doesn't support the latest graphics cards

**Comment by wuzhiwei2001** [comment]
> Maybe you should update the torch version, torch 1.6+cuda10.1 doesn't support the latest graphics cards

Thank you very much for your reply! I can run it now!

**Comment by MrAllenaz** [comment]
> I trained 70 epochs but the results are still bad, including the errors and the loss. The loss is always like 33, 34..., is it normal or something goes wrong? ![微信图片_20240712155952](https://private-user-images.githubusercontent.com/118343549/348173689-fae57e08-5c87-4aab-8f73-e54bea819b4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MTE5NzQsIm5iZiI6MTcyNTYxMTY3NCwicGF0aCI6Ii8xMTgzNDM1NDkvMzQ4MTczNjg5LWZhZTU3ZTA4LTVjODctNGFhYi04ZjczLWU1NGJlYTgxOWI0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwNlQwODM0MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ODlmNjRkNWRjN2M1MDk3OGM1MzY3OTExMWI3ODg3YmZhODQ0Mzg3Y2QzY2RlYzY2YmZiYjgxMmY0NmY4YTZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.0QAleCXu4QwOC1m4F4sQ1hWs4B_ge1X4EThBfbyIhQQ)

Have you solved this problem yet? I have trained 124epoch using a single GPU , but the results are still bad.I'm wondering if it's because the batchsize is too small.
![image](https://github.com/user-attachments/assets/2432c506-54d9-4ae9-9e78-ed4f7a1018ec)

**Comment by A11en4z** [comment]
I tried single GPU, dual GPU, batch=2,4 respectively and none of them could train effectively. But when I use 4 GPUs, batch=2 there is no case of Loss not decreasing.

**Comment by wuzhiwei2001** [comment]
> I tried single GPU, dual GPU, batch=2,4 respectively and none of them could train effectively. But when I use 4 GPUs, batch=2 there is no case of Loss not decreasing.

How much does your loss have to decrease before it stops decreasing? thanks!

**Comment by A11en4z** [comment]
> > I tried single GPU, dual GPU, batch=2,4 respectively and none of them could train effectively. But when I use 4 GPUs, batch=2 there is no case of Loss not decreasing.
> 
> How much does your loss have to decrease before it stops decreasing? thanks!

Before I used 4gpu the loss would float around 32, 33 and the accuracy was poor, I then tried the weights posted by the author on a subset and the loss was around 15, you can use this as a reference

**Comment by wuzhiwei2001** [comment]
> > > I tried single GPU, dual GPU, batch=2,4 respectively and none of them could train effectively. But when I use 4 GPUs, batch=2 there is no case of Loss not decreasing.
> > 
> > 
> > How much does your loss have to decrease before it stops decreasing? thanks!
> 
> Before I used 4gpu the loss would float around 32, 33 and the accuracy was poor, I then tried the weights posted by the author on a subset and the loss was around 15, you can use this as a reference

thanks!

**Comment by iamagoodboyyeyeye** [comment]
> 我训练了 70 个 epoch，但结果仍然很糟糕，包括错误和损失。损失总是像 33、34......，这是正常的还是出了什么问题？ ![微信图片_20240712155952](https://private-user-images.githubusercontent.com/118343549/348173689-fae57e08-5c87-4aab-8f73-e54bea819b4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0OTc0MzcsIm5iZiI6MTcyOTQ5NzEzNywicGF0aCI6Ii8xMTgzNDM1NDkvMzQ4MTczNjg5LWZhZTU3ZTA4LTVjODctNGFhYi04ZjczLWU1NGJlYTgxOWI0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMVQwNzUyMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNzU5YWNmMmM0NTUwMmNkMGRjOGZkYTdlNGM5OWFhODlkMDcwMWY0MWMwYjhlNjQ5YzcyOGU4YjczMmVhNTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KkDpbBsluRAFLKB6qaibr0z5azMgq-dIQxl_3Z-CKkg)

Hello, may I ask how many GPUs you used for training and have you produced the results now? I also encountered this problem with a GPU

**Comment by iamagoodboyyeyeye** [comment]
> 我分别尝试了单 GPU、双 GPU、batch=2,4，但它们都无法有效训练。但是当我使用 4 个 GPU、batch=2 时、没有 Loss 不减少的情况。

I also have problems using a single GPU, so is it a problem with the number of GPUs used?

**Comment by A11en4z** [comment]
> > 我训练了 70 个 epoch，但结果仍然很糟糕，包括错误和损失。损失总是像 33、34......，这是正常的还是出了什么问题？ ![微信图片_20240712155952](https://private-user-images.githubusercontent.com/118343549/348173689-fae57e08-5c87-4aab-8f73-e54bea819b4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0OTc0MzcsIm5iZiI6MTcyOTQ5NzEzNywicGF0aCI6Ii8xMTgzNDM1NDkvMzQ4MTczNjg5LWZhZTU3ZTA4LTVjODctNGFhYi04ZjczLWU1NGJlYTgxOWI0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMVQwNzUyMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNzU5YWNmMmM0NTUwMmNkMGRjOGZkYTdlNGM5OWFhODlkMDcwMWY0MWMwYjhlNjQ5YzcyOGU4YjczMmVhNTg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KkDpbBsluRAFLKB6qaibr0z5azMgq-dIQxl_3Z-CKkg)
> 
> Hello, may I ask how many GPUs you used for training and have you produced the results now? I also encountered this problem with a GPU

Hello,I remember I was using 4 GPUs and could train normally, but I didn't run it on the entire VG training set because it took too long to train

**Comment by A11en4z** [comment]
> > 我分别尝试了单 GPU、双 GPU、batch=2,4，但它们都无法有效训练。但是当我使用 4 个 GPU、batch=2 时、没有 Loss 不减少的情况。
> 
> I also have problems using a single GPU, so is it a problem with the number of GPUs used?

hello,I tried 2 GPUs and increasing the batchsize and it didn't work, but I was able to train fine with 4 GPUs

**Comment by iamagoodboyyeyeye** [comment]
> > > 我分别尝试了单 GPU、双 GPU、batch=2,4，但它们都无法有效训练。但是当我使用 4 个 GPU、batch=2 时、没有 Loss 不减少的情况。
> > 
> > 
> > I also have problems using a single GPU, so is it a problem with the number of GPUs used?
> 
> hello,I tried 2 GPUs and increasing the batchsize and it didn't work, but I was able to train fine with 4 GPUs

Thank you for your reply. If possible, I would like to add a WeChat account to communicate. My WeChat ID is zhl15042182325
